정민지/mvvmstudy/app/src/main/java/com/github/sookhee/mvvmstudy/ui/main/MainViewModel.kt

```diff
@@ -0,0 +1,62 @@
+package com.github.sookhee.mvvmstudy.ui.main
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.github.sookhee.mvvmstudy.ResultState
+import com.github.sookhee.mvvmstudy.model.GithubRepositoryModel
+import com.github.sookhee.mvvmstudy.network.GithubAPI
+import com.github.sookhee.mvvmstudy.network.RetrofitClient
+import com.github.sookhee.mvvmstudy.network.spec.GithubRepositoryResponse
+import com.github.sookhee.mvvmstudy.repository.GithubRepository
+import com.github.sookhee.mvvmstudy.repository.OnNetworkCallbackListener
+
+/**
+ *  MainViewModel.kt
+ *
+ *  Created by Minji Jeong on 2022/05/02
+ *  Copyright © 2022 GwanakMT All rights reserved.
+ */
+
+class MainViewModel : ViewModel(), OnNetworkCallbackListener {
+    private val _repositoryResultState = MutableLiveData<ResultState>()
+    val repositoryResultState: LiveData<ResultState> = _repositoryResultState
+
+    private val request by lazy { RetrofitClient.buildService(GithubAPI::class.java) }
+    private val githubRepository = GithubRepository(request, this)
+
+    override fun onSuccess(list: List<GithubRepositoryResponse>) {
+        _repositoryResultState.postValue(ResultState.Success(mapToGithubRepositoryModelList(list)))
+    }
+
+    override fun onFailure(throwable: Throwable) {
+        _repositoryResultState.postValue(ResultState.Error(throwable.message.toString()))
+    }
+
+    fun requestDataToGithubAPI(keyword: String) {
+        _repositoryResultState.postValue(ResultState.Loading)
+        if (keyword.isNotEmpty()) {
+            githubRepository.getGithubRepositoryListWithQuery(keyword)
+        } else {
+            githubRepository.getGithubRepositoryList()
+        }
+    }
+
+    private fun mapToGithubRepositoryModelList(response: List<GithubRepositoryResponse>): List<GithubRepositoryModel> {
+        val list = mutableListOf<GithubRepositoryModel>()
+        response.forEach {
+            list.add(
+                GithubRepositoryModel(
+                    id = it.id,
+                    repoName = it.name,
+                    repoLastUpdate = it.lastUpdate ?: "",
+                    language = it.language ?: "",
+                    ownerName = it.owner.name,
+                    profileImage = it.owner.profileImage
+                )
+            )
+        }
+
+        return list
+    }
```

Comment:
list의 map를 사용해도 좋을 것같아요!
```suggestion
       return response.map { repository ->
           GithubRepositoryModel(
                    id = repository.id,
                    repoName = repository.name,
                    repoLastUpdate = repository.lastUpdate ?: "",
                    language = repository.language ?: "",
                    ownerName = repository.owner.name,
                    profileImage = repository.owner.profileImage
                )
       }
    }
```

Reply:
헉 예제까지.. 감사합니당